src/Microsoft.VisualStudio.Threading/DispatcherExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿/********************************************************
+*                                                        *
+*   © Copyright (C) Microsoft. All rights reserved.      *
+*                                                        *
+*********************************************************/
+
+#if NET45
+
+namespace Microsoft.VisualStudio.Threading
+{
+    using System;
+    using System.Threading;
+    using System.Windows.Threading;
+
+    /// <summary>
+    /// Extension methods for the WPF <see cref="Dispatcher"/> for better
+    /// interop with the <see cref="JoinableTaskFactory"/>.
+    /// </summary>
+    public static class DispatcherExtensions
+    {
+        /// <summary>
+        /// Creates a <see cref="JoinableTaskFactory"/> that schedules work with the specified
+        /// <see cref="Dispatcher"/> and <see cref="DispatcherPriority"/>.
+        /// </summary>
+        /// <param name="joinableTaskFactory">The underlying <see cref="JoinableTaskFactory"/> to use.</param>
+        /// <param name="priority">
+        /// The priority with which to schedule any work on the UI thread,
+        /// when and if <see cref="JoinableTaskFactory.SwitchToMainThreadAsync"/> is called.
+        /// </param>
+        /// <param name="dispatcher">The <see cref="Dispatcher"/> that schedules work on the main thread.</param>
+        /// <returns>A <see cref="JoinableTaskFactory"/> that may be used for scheduling async work with the specified priority.</returns>
+        public static JoinableTaskFactory WithPriority(this JoinableTaskFactory joinableTaskFactory, DispatcherPriority priority, Dispatcher dispatcher)
```

Comment:
💭 It "feels weird" having the arguments in this order. Seems like `dispatcher` should be before `priority`.